packages/block-editor/src/components/list-view/block.js

```diff
@@ -324,6 +332,19 @@ function ListViewBlock( {
 			collapseAll();
 			// Expand all parents of the current block.
 			expand( blockParents );
+		} else if ( isMatch( 'core/block-editor/group', event ) ) {
+			const clientIds = getSelectedBlockClientIds();
```

Comment:
In the List View, rather than using the selected block client ids, it's generally better to use `const { blocksToUpdate } = getBlocksToUpdate();` as used in the insert before and after keyboard shortcuts above. This will ensure that if the user is focused on a block that isn't part of the selection, then the action is performed on the currently focused block. If the focused block is part of the selection, then the whole selection is used.

Reply:
I just realised this might not be particularly clear as to why it matters. Here's a quick example video of the behaviour as it is currently, where if I move up the list view to outside the selection, the keyboard shortcut acts on the selected blocks instead of the focused one:

https://github.com/WordPress/gutenberg/assets/14988353/f66c745c-e4e4-44ad-b5e6-7786833a1382